webknossos/tests/test_cli.py

```diff
@@ -0,0 +1,477 @@
+        wkw_path = Path("wkw_from_tiff_extended")
+
+        result = runner.invoke(
+            app,
+            [
+                "convert",
+                "--voxel-size",
+                "11.0,11.0,11.0",
+                "--data-format",
+                "wkw",
+                "--name",
+                "wkw_from_tiff",
+                "--compress",
+                str(origin_path),
+                str(wkw_path),
+            ],
+        )
+
+        assert result.exit_code == 0
+        assert (wkw_path / PROPERTIES_FILE_NAME).exists()
+
+
+@pytest.mark.parametrize(
+    "url",
+    [
+        "https://webknossos.org/datasets/scalable_minds/cremi_example/",
+        "https://webknossos.org/datasets/scalable_minds/cremi_example/view#512,512,16,0,1.3",
+        "https://webknossos.org/links/upcKUKDe5CatK4JX",
+    ],
+)
+@pytest.mark.block_network(allowed_hosts=[".*"])
+@pytest.mark.vcr(ignore_hosts=["webknossos.org", "data-humerus.webknossos.org"])
+def test_download_dataset(url: str) -> None:
+    """Tests the functionality of download subcommand."""
+
+    result = runner.invoke(app, ["download"])
+    assert result.exit_code == 2
+
+    with tmp_cwd():
+        result = runner.invoke(
+            app,
+            [
+                "download",
+                "--bbox",
+                "0,0,0,5,5,5",
+                "--mag",
+                "8",
+                "--url",
+                url,
+                "testoutput/",
+            ],
+        )
+        assert result.exit_code == 0
+        assert (Path("testoutput") / PROPERTIES_FILE_NAME).exists()
+
+
+@pytest.mark.filterwarnings("ignore::UserWarning")
+def test_downsample_and_upsample() -> None:
+    """Tests the functionality of downsample subcommand."""
+
+    result_without_args = runner.invoke(app, ["downsample"])
+    assert result_without_args.exit_code == 2
+
+    with tmp_cwd():
+        wkw_path = Path("simple_wkw_dataset")
+        copytree(TESTDATA_DIR / wkw_path, wkw_path)
+
+        result_downsample = runner.invoke(app, ["downsample", str(wkw_path)])
+
+        assert result_downsample.exit_code == 0
+        assert (wkw_path / "color" / "1" / "z0" / "y0" / "x0.wkw").exists()
+        assert (wkw_path / "color" / "2" / "z0" / "y0" / "x0.wkw").exists()
+        assert (wkw_path / "color" / "4" / "z0" / "y0" / "x0.wkw").exists()
+
+        Dataset.open(wkw_path).get_layer("color").delete_mag(1)
+
+        assert not (wkw_path / "color" / "1" / "z0" / "y0" / "x0.wkw").exists()
+
+        result_upsample = runner.invoke(
+            app, ["upsample", "--from-mag", "2", str(wkw_path)]
+        )
+
+        assert result_upsample.exit_code == 0
+        assert (wkw_path / "color" / "1" / "z0" / "y0" / "x0.wkw").exists()
+
+
+def test_upload() -> None:
+    """Tests the functionality of upload subcommand."""
+
+    result_without_args = runner.invoke(app, ["upload"])
+    assert result_without_args.exit_code == 2
+
+
+def test_upsample() -> None:
```

Comment:
I think you can remove this, since the upsampling is already tested in test_downsample_and_upsample